ably/auth_test.go

```diff
@@ -10,11 +10,11 @@ import (
 var _ = Describe("Auth", func() {
 	Describe("RequestToken", func() {
 		It("gets a token from the API", func() {
-			req := client.Auth.CreateTokenRequest()
+			req := client.Auth().CreateTokenRequest()
```

Comment:
> The only way for a property to be immutable outside the package is to use accessors.

This is not necessarily true, the same problem is in Java - even though you return a variable with accessor without defensive copying it's still can be modified. So we gain nothing with using accessors. However I very like using accessor vs exported fields (exported fields should be used only for model, not control imho, sorry for offtop), as then I can make them being constructed lazily, which allows for using zero-value `RestClient`, which is preferable in Go.

Comming back to `Presence`, the `RestPresence` does not have any fields exported, so the user can't modify it either way :)